src/Symfony/Component/Form/Form.php

```diff
@@ -508,22 +508,24 @@ public function submit($submittedData, bool $clearMissing = true)
         $dispatcher = $this->config->getEventDispatcher();
 
         // Obviously, a disabled form should not change its data upon submission.
-        if ($this->isDisabled() && $this->isRoot()) {
+        if ($this->isDisabled()) {
             $this->submitted = true;
 
-            if ($dispatcher->hasListeners(FormEvents::PRE_SUBMIT)) {
-                $event = new FormEvent($this, $submittedData);
-                $dispatcher->dispatch(FormEvents::PRE_SUBMIT, $event);
-            }
+            if ($this->isRoot()) {
+                if ($dispatcher->hasListeners(FormEvents::PRE_SUBMIT)) {
+                    $event = new FormEvent($this, $submittedData);
+                    $dispatcher->dispatch($event, FormEvents::PRE_SUBMIT);
+                }
 
-            if ($dispatcher->hasListeners(FormEvents::SUBMIT)) {
-                $event = new FormEvent($this, $this->getNormData());
-                $dispatcher->dispatch(FormEvents::SUBMIT, $event);
-            }
+                if ($dispatcher->hasListeners(FormEvents::SUBMIT)) {
+                    $event = new FormEvent($this, $this->getNormData());
+                    $dispatcher->dispatch($event, FormEvents::SUBMIT);
```

Comment:
I've added [another test](https://github.com/symfony/symfony/pull/38025/files#diff-de65be85a5affed1f499b79d96965419R74-R94) covering this case where the underlying object has been changed. This shouldn't happen if the entire form is disabled. Also this would break BC.